pkg/controller/logstorage/users/users_controller.go

```diff
@@ -357,15 +358,51 @@ func (r *UsersCleanupController) Reconcile(ctx context.Context, request reconcil
 		request.Namespace, "Request.Name", request.Name, "installNS", helper.InstallNamespace(), "truthNS", helper.TruthNamespace())
 	reqLogger.Info("Reconciling LogStorage - Cleanup")
 
-	// Wait for Elasticsearch to be installed and available.
-	elasticsearch, err := utils.GetElasticsearch(ctx, r.client)
-	if err != nil {
+	// We skip requests without a namespace specified in multi-tenant setups.
+	if r.multiTenant && request.Namespace == "" {
+		return reconcile.Result{}, nil
+	}
+
+	// Check if this is a tenant-scoped request.
+	_, _, err := utils.GetTenant(ctx, r.multiTenant, r.client, request.Namespace)
```

Comment:
Is this right? Shouldn't we still reconcile if there is no tenant since we need to clean up users for that tenant?

Reply:
That's probably a good point, I don't know that it means that this should change here though. We probably want to ensure the tenant resource isn't actually deleted until it's cleaned up after. Otherwise we can't know if the request for a tenant or because of change in a common namespace.

Reply:
I think this branch is effectively dead code, thinking about it some more.

We don't watch anything in a common namespace for this controller: https://github.com/tigera/operator/blob/191bc39643441e935e09ff9b544da607f9627ec6/pkg/controller/logstorage/users/users_controller.go#L145-L158

I think we should just remove this check - seems like the presence of this code is misleading, and negates the "periodic reconcile" behavior that we want.

Reply:
I suppose we probably want a watch for LogStorage to be added, if we are going to add that check here ( seems like we probably should, but also it's probably not a huge issue).

If we do, I think it is reasonable to reconcile users when the log storage changes.

Reply:
Personally I'm a little hesitant about not handling the case where we watching something from a different namespace, even if it's not currently done.

I think the assumptions for these controllers have been you can watch what you need to from any namespace, so there's really no reason why this couldn't change, then we do have a bug.

Unless we have some sort of standard that this is a specific type of controller that can only ever watch specific namespaces (and also not global values), and we also enforce this (possible by wrapping the controller / watch registration to fail if the correct namespace isn't added to the watch), then this seems like a footgun.

Reply:
> Personally I'm a little hesitant about not handling the case where we watching something from a different namespace, even if it's not currently done.

The design of this controller is that it is a full reconcile on every event, regardless of what triggered it. I believe that to mean that the controller already properly handles events triggered from other namespaces. I am against adding code that is effectively unused, on the premise that it might someday be needed - if we add such a watch in the future, then we can add the necessary logic to the body of Reconcile() at the same time (which may or may not involve a tenant check depending on what the actual change is). 

Specifically adding this check breaks the periodic reconcile behavior of this controller, which is important, so we need to at least address that before we can merge this change.

Reply:
Either way, I think it should be a second PR - seems unrelated to the fix at hand here.